File: agents/entrypoint.py
from __future__ import annotations

import asyncio
import os
import socket

from agents.config.settings import Settings
from agents.leader.leader_agent import LeaderAgent
from agents.worker.worker_agent import WorkerAgent


async def _amain() -> None:
    role = os.environ.get("AGENT_ROLE", "worker")
    hostname = socket.gethostname()
    agent_id = os.environ.get("AGENT_ID") or f"{role}-{hostname[:8]}"

    settings = Settings.from_env()
    if role == "orchestrator":
        # Run the per-job crew orchestrator.
        from agents.orchestrator.orchestrator import main as orch_main

        await orch_main()
        return
    if role == "leader":
        agent = LeaderAgent(agent_id=agent_id, settings=settings)
    else:
        agent = WorkerAgent(agent_id=agent_id, settings=settings)

    await agent.start()


def main() -> None:
    asyncio.run(_amain())


if __name__ == "__main__":
    main()

File: packages/agents/src/entrypoint.ts
import os from "node:os";
import { settingsFromEnv } from "./config/settings.js";
import { LeaderAgent } from "./leader/leader-agent.js";
import { WorkerAgent } from "./worker/worker-agent.js";

async function main(): Promise<void> {
  const role = process.env.AGENT_ROLE ?? "worker";
  const hostname = os.hostname();
  const agentId = process.env.AGENT_ID ?? `${role}-${hostname.slice(0, 8)}`;

  const settings = settingsFromEnv();

  if (role === "orchestrator") {
    const { main: orchMain } = await import(
      "./orchestrator/orchestrator.js"
    );
    await orchMain();
    return;
  }

  if (role === "leader") {
    const agent = new LeaderAgent(agentId, settings);
    await agent.start();
  } else {
    const agent = new WorkerAgent(agentId, settings);
    await agent.start();
  }
}

main().catch((err) => {
  console.error("[entrypoint] fatal:", err);
  process.exit(1);
});
